pkg/lib/bundle/validate.go

```diff
@@ -170,6 +192,64 @@ func (i imageValidator) ValidateBundleFormat(directory string) error {
 	return nil
 }
 
+func validateProperties(manifestsDir string, fileBasedProperties []PropertiesTypeVal) ([]error, []string) {
+	var validationErrors []error
+	var validationWarnings []string
+
+	// Read the manifests directory for the CSV
+	manifestFiles, err := ioutil.ReadDir(manifestsDir)
+	if err != nil {
+		validationErrors = append(validationErrors, fmt.Errorf("Failed to read manifests directory while validating properties: %w", err))
+	}
+
+	// Find the CSV
+	csvFound := false
+	csvFile := &v1.ClusterServiceVersion{}
+	for _, f := range manifestFiles {
+		err = registry.DecodeFile(filepath.Join(manifestsDir, f.Name()), &csvFile)
+		if err == nil && csvFile.Kind == CSVKind {
+			csvFound = true
+			break
+		}
+	}
+
+	// If CSV is found and it has the olm.properties attribute defined
+	if csvFound {
+		if annotationPropertiesString, defined := csvFile.Annotations["olm.properties"]; defined {
+			// Structure the olm.properties into a comparable format
+			var annotationBasedProperties []PropertiesTypeVal
+			err := json.Unmarshal([]byte(annotationPropertiesString), &annotationBasedProperties)
+			if err != nil {
+				validationErrors = append(validationErrors, fmt.Errorf("Failed to read CSV file while validating properties: %w", err))
+			}
+
+			// Determine which props have more values
+			moreProperties := annotationBasedProperties
+			lessProperties := fileBasedProperties
+			if len(moreProperties) < len(lessProperties) {
+				moreProperties = fileBasedProperties
+				lessProperties = annotationBasedProperties
+			}
+
+			// For each property in both slices, determine if a property is duplicated, then warn if not done already
+			alreadyWarned := map[string]bool{}
```

Comment:
If you're only looking at the property types and not checking for multiple values, you can also keep a separate map of types from one property set and validate it against the other.

Reply:
Check the most recent push, I believe it will cover this issue.